.github/workflows/python-package.yml

```diff
@@ -0,0 +1,40 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Python package
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ['3.7', '3.8']
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        pip install --upgrade pip pytest
+        pip install --upgrade git+https://github.com/aws/amazon-braket-schemas-python@main
+        pip install --upgrade git+https://github.com/aws/amazon-braket-default-simulator-python@main
+        pip install -e .[test]
+    - name: Check code format
+      run: |
+        # stop the build if there are Python format errors or undefined names
+        black --check .
+        flake8
+    - name: Run unit tests
+      run: |
+        pytest test/unit_tests
```

Comment:
* Is there a limitation to running these steps through tox ? Since our recommendation for running these checks locally is through tox, it might be helpful to align the two. 
* A code coverage check might be a useful addition here.

Reply:
I'm just migrating over from CodeBuild, where these are the steps used; can't use tox for linters quite yet, as the `linters` step in tox.ini also changes the code, which we don't want; we only want to _check_ the format. As for coverage, I'm planning to onboard to [CodeCov](https://about.codecov.io/).